sources/osgShader/ShaderGenerator.js

```diff
@@ -251,9 +272,62 @@ ShaderGenerator.prototype = {
                 this.getActiveAttributeListCache(state) +
                 this.getActiveTextureAttributeListCache(state);
 
-            var cache = this._cache[hash];
-            if (cache !== undefined) return cache;
+            var cachedProgram = this._cache[hash];
+            if (cachedProgram !== undefined && !cachedProgram.getAsyncCompiling()) {
+                return cachedProgram;
+            }
 
+            var waitingProgram;
+            var asyncShader = true;
+            var synchronizedCompilation = !asyncShader;
+            if (!synchronizedCompilation) {
```

Comment:
not sure I get the tree lines above, so !synchronizedCompilation is always true?

Reply:
it's a global for debug, comparing both async and sync.